### ISSUE TITLE
Make SCI dependency optional for smaller JS builds

### ISSUE THREAD

**Issue by Naomarik** [metosin/malli#203]
Currently in the context of my application I will never have a need for it and it adds 93kb gzipped at compression level 6.

**Comment by ikitommi** [comment]
Did a quick test using shadow-cljs:

```clj
metosin/malli 43.1 KB
borkdude/sci  63.6 KB
```

The malli is currently not optimized either for bundle size. Will make sci optional and try to make malli smaller too.

**Comment by ikitommi** [comment]
Added shadow-cljs as dev-dependency and guide how to measure the bundle sizes. With the latest versions (sci got docstrings, making it a lot bigger):

```
metosin/malli 50KB -> 11.8KB GZIP
borkdude/sci  564KB - > 120KB GZIP
```

**Comment by ikitommi** [comment]
Not sure how correctly the bundle size is calculated, but here goes the report:

<img width="1052" alt="Screenshot 2020-06-12 at 8 54 51" src="https://user-images.githubusercontent.com/567532/84469881-c04da600-ac8a-11ea-8f03-66d5ca81381c.png">

**Comment by borkdude** [comment]
Work in progress at #210. It works, but the tests need to be refactored to account for the optionality of sci.

**Comment by borkdude** [comment]
I have closed the PR. Instead, I'm offering you this library https://github.com/borkdude/dynaload. If you need any help, let me know. :)

**Comment by ikitommi** [comment]
thanks, will try that :)

**Comment by ikitommi** [comment]
Will be fixed in #227 .

**Comment by ikitommi** [comment]
sci is now disabled by default. README has guide how to enable it for cljs. With the minimal dummy sample app:

* js: 757,8kb -> 136kb
* gzipped: 161kb -> 31,5kb

# with sci:

<img width="911" alt="with-sci" src="https://user-images.githubusercontent.com/567532/88318717-d340a500-cd23-11ea-88ae-08e86600ad69.png">

# without (default):

<img width="910" alt="no-sci" src="https://user-images.githubusercontent.com/567532/88318733-d9cf1c80-cd23-11ea-8673-d222407839bd.png">
